scripts/delete-account.js

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env node
+
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/*/
+
+ Usage:
+
+ node scripts/delete-account.js user@example.com
+
+ This script is used to manually delete a user's account,
+ e.g. in response to a GDPR deletion request.  It uses the
+ same config file as key_server.js so should be run from a
+ production instance, and it tries to re-use as much of the
+ actual delete-account route code as possible.
+
+/*/
+
+'use strict'
+
+const readline = require('readline')
+const P = require('../lib/promise')
+const config = require('../config').getProperties()
+const log = require('../lib/log')(config.log.level)
+const Token = require('../lib/tokens')(log, config)
+const error = null
+const mailer = null
+const MockPassword = null
+const mockCustoms = { check: () => {} }
+const push = null
+
+
+var DB = require('../lib/db')(
+  config,
+  log,
+  Token
+)
+
+DB.connect(config[config.db.backend]).then(db => {
+
+  // Bypass password checks.
+  function MockPassword() { }
+  MockPassword.prototype.verifyHash = () { return P.resolve(true) }
+
+  // Bypass TOTP checks.
+  db.totpToken = () => { return P.resolve(false) }
+
+  const signinUtils = require('../lib/routes/utils/signin')(log, config, mockCustoms, db, mailer)
+  const accountDestroyRoute = require('../lib/routes/account')(
+    log,
+    db,
+    mailer,
+    MockPassword,
+    config,
+    mockCustoms,
+    signinUtils,
+    push
+  ).find(r => r.path === '/account/destroy')
+
+
+  P.each(process.argv.slice(1), email => {
+    return db.accountRecord(email).then(account => {
+      console.log("Found account record:")
+      console.log("    uid:", account.uid)
+      console.log("    email:", account.email)
+      const rl = readline.createInterface({
+        input: process.stdin,
+        output: process.stdout,
+      })
+      return new Promise((resolve, reject) => {
```

Comment:
I think in practice it works out okay but looks like you're mixing a native promise in with bluebird stuff here, is that what you wanted to do?

Reply:
not what I wanted; thanks!